File: app/src/main/java/universal/appfactory/aeroindia2023/ExampleNotificationReceivedHandler.java
package universal.appfactory.aeroindia2023;

import android.content.Context;
import android.util.Log;
import com.onesignal.OSNotificationReceivedEvent;
import com.onesignal.OneSignal;

import org.json.JSONObject;

class ExampleNotificationReceivedHandler implements OneSignal.OSRemoteNotificationReceivedHandler {

    @Override
    public void remoteNotificationReceived(Context context, OSNotificationReceivedEvent osNotificationReceivedEvent) {
        JSONObject data = osNotificationReceivedEvent.getNotification().getAdditionalData();
        String customKey;

        if (data != null) {
            customKey = data.optString("customkey", null);
            Log.i("JSON String", data.toString());
            Log.i("OneSignalExample", "custom key set with value: " + customKey);
        }
        else{
            Log.i("OneSignalExample", "No message passed");
        }
    }
}

File: app/src/main/java/universal/appfactory/aeroindia2023/ExampleNotificationReceivedHandler.kt
import android.app.Application
import android.util.Log
import com.onesignal.OSNotification
import com.onesignal.OSNotificationReceivedEvent
import com.onesignal.OneSignal


internal class OSNotificationReceivedHandler(

    private val application: Application

) : OneSignal.OSNotificationWillShowInForegroundHandler {


    companion object {
        private const val TAG = "OneSignalReceivedHandler"
    }


    fun notificationReceived(notification: OSNotification) {

        val data = notification.additionalData

        val notificationID = notification.notificationId
        val title = notification.title
        val body = notification.body
        // ...

        Log.i(TAG, "NotificationID received: $notificationID")
        Log.i("Message title", title.toString())

    }

    override fun notificationWillShowInForeground(event: OSNotificationReceivedEvent?) {
        OneSignal.onesignalLog(
            OneSignal.LOG_LEVEL.VERBOSE, "NotificationWillShowInForegroundHandler fired!" +
                    " with notification event: " + event.toString()
        )

        val notification: OSNotification = event!!.notification
        val data = notification.additionalData

        Log.i("JSON data", data.toString())

        // Complete with null means don't show a notification.
        event.complete(notification)

    }

}